volatility3/framework/plugins/linux/proc.py

```diff
@@ -35,16 +40,138 @@ def get_requirements(cls):
                 element_type=int,
                 optional=True,
             ),
+            requirements.BooleanRequirement(
+                name="dump",
+                description="Extract listed memory segments",
+                default=False,
+                optional=True,
+            ),
+            requirements.ListRequirement(
+                name="address",
+                description="Process virtual memory address to include "
```

Comment:
I'm a little nervous about this.  The description needs to be extremely precise so that no one gets confused.  It should be clear whether this value applies to all output or just dumping, and it should be clear whether we're discussing addresses or ranges (since the name is address, but it then after says all other ranges are excluded).  It sounds as though this is a single address, but base address isn't clear whether it's only the base address of the module or the base (starting) address of a block?  Also, it feels as though there could be some clever code in the plugin that *does* allow any address in a block to be used, which might be handier for people who want to extract just a chunk they're interested in, without having to do extra work to figure out the starting address of the segment they're interested in?

Reply:
I think I know how I'd add the any address option. I'll update when that is done.